### PR TITLE
Backport of Windows exception handler from Python 3.5 + adjust tests

### PR DESCRIPTION
**Pull request by nirgilboa**
Original patch found at: https://bugs.python.org/issue23848

### CODE REVIEW

File: faulthandler.c [hunk 1]
```diff
@@ -967,28 +1019,34 @@ Py_uintptr_t
 stack_overflow(Py_uintptr_t min_sp, Py_uintptr_t max_sp, size_t *depth)
 {
     /* allocate 4096 bytes on the stack at each call */
-    unsigned char buffer[4096];
+    unsigned char buffer[1024*1024];
     Py_uintptr_t sp = (Py_uintptr_t)&buffer;
     *depth += 1;
-    if (sp < min_sp || max_sp < sp)
+    if (sp < min_sp || max_sp < sp) {
+        printf("call #%lu\n", (unsigned long)*depth);
```
Comment (haypo) [inline faulthandler.c]: Please remove the printf.

**Comment by haypo** [comment]
I'm in holiday, ping me back in one week if you want a new review ;-)

File: faulthandler.c (hunk 1 above)
Comment (nirgilboa) [reply faulthandler.c]: Removed it by amending the commit so as to keep the pull request clean

**Comment by nirgilboa** [comment]
Ping :)
Hope you had a nice holiday - I pushed an amended commit, removing the printf.

**Comment by haypo** [comment]
Thanks for your backport. I just fixed a mistake your made in the initfaulthandler() function. It's now fixed (see my following commit).

I just released faulthandler 3.0 with your backport: https://faulthandler.readthedocs.io/#version-3-0-2017-09-16

**Comment by haypo** [comment]
Oh, now I recall why a faulthandler is painful: I have to build wheel packages for Windows. Ok, now the 3.0 is release is really complete :-) I built 32-bit and 64-bit wheel packages for Windows.

**Comment by nirgilboa** [comment]
Great ! Thank you! :-)

**Pull request merged**